### PR TITLE
Add Checking and Force Flag for Directory in Serve Command

### PR DESCRIPTION
**Pull request by SquirrelHub**
Attempts to address #2251 by mimicking behavior of `build` command's checking of existing folders from the `--output-dir` flag, and also allows the use of a new `--force` flag to override said directory for serving the files.

**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

File: docs/content/documentation/getting-started/cli-usage.md [hunk 1]
```diff
@@ -81,6 +81,8 @@ web browser.
 
 Before starting, Zola will delete the output directory (by default `public` in project root) to start from a clean slate.
 
+If you are specifying the directory however using the `output-dir` flag, Zola will not use the specified directory if it already exists unless the --force flag is used.
```
Comment (Keats) [inline docs/content/documentation/getting-started/cli-usage.md]: could be just my English but wouldn't `If you are however specifying the director using the `output-dir` flag` be better?

File: src/cmd/serve.rs [hunk 2]
```diff
@@ -268,6 +269,12 @@ fn create_new_site(
     site.enable_serve_mode();
     site.set_base_url(base_url);
     if let Some(output_dir) = output_dir {
+        if !force && output_dir.exists() {
```
Comment (Keats) [inline src/cmd/serve.rs:272]: I would swap the 2 conditions just because it would be clearer imo

**Review by Keats — APPROVED** [review]
Just some minor tweaks if you want, nothing big

File: docs/content/documentation/getting-started/cli-usage.md (hunk 1 above)
Comment (Wabuo) [reply docs/content/documentation/getting-started/cli-usage.md]: Randomly came across this while checking out Zola ... I'm throwing
``If you are specifying the directory but are also using the `output-dir` flag, Zola will not use the specified directory if it already exists unless the --force flag is used.``
in the ring.

File: src/cmd/serve.rs (hunk 2 above)
Comment (SquirrelHub) [reply src/cmd/serve.rs:272]: This pattern honestly was taken straight from [the build command's](https://github.com/getzola/zola/blob/695c17d2222e2ee3f83d304d0828265b1c7407cf/src/cmd/build.rs#L17) pattern of ifs. 

Do you still think it's worth changing?

File: docs/content/documentation/getting-started/cli-usage.md (hunk 1 above)
Comment (SquirrelHub) [reply docs/content/documentation/getting-started/cli-usage.md]: I used the wording that @Wabuo provided. Let me know if it looks alright @Keats.

File: src/cmd/serve.rs (hunk 2 above)
Comment (Keats) [reply src/cmd/serve.rs:272]: Nah it's fine

**Review by Keats — APPROVED** [review]

**Pull request merged**